pkg/pipelinectxt/artifacts.go

```diff
@@ -203,46 +214,53 @@ func ArtifactGroup(ctxt *ODSContext, subdir string) string {
 
 // DownloadGroup searches given repositories in order for assets in given group.
 // As soon as one repository has any asset in the group, the search is stopped
-// and all fond artifacts are downloaded into artifactsDir.
+// and all found artifacts are downloaded into artifactsDir.
 // An artifacts manifest is returned describing the downloaded files.
 // When none of the given repositories contains any artifacts under the group,
 // no artifacts are downloaded and no error is returned.
-func DownloadGroup(nexusClient nexus.ClientInterface, repository string, group, artifactsDir string, logger logging.LeveledLoggerInterface) (*ArtifactsManifest, error) {
+// If artifactsDir is an empty string, the searched files are downloaded but
+// the artifacts are still recorded in the returned manifest.
+func DownloadGroup(
+	nexusClient nexus.ClientInterface,
+	repository, group, artifactsDir string,
+	logger logging.LeveledLoggerInterface) (*ArtifactsManifest, error) {
 	// We want to target all artifacts underneath the group, hence the trailing '*'.
 	nexusSearchGroup := fmt.Sprintf("%s/*", group)
-	am := &ArtifactsManifest{
-		Artifacts: []ArtifactInfo{},
-	}
+	am := NewArtifactsManifest(repository)
 	urls, err := searchForAssets(nexusClient, nexusSearchGroup, repository, logger)
 	if err != nil {
 		return nil, err
 	}
 
+	if artifactsDir == "" {
+		logger.Debugf("Artifacts will not be downloaded but only added to the manifest ...")
+	}
+
 	for _, s := range urls {
 		u, err := url.Parse(s)
 		if err != nil {
 			return nil, err
 		}
-		urlPathParts := strings.Split(u.Path, fmt.Sprintf("%s/", group))
-		if len(urlPathParts) != 2 {
-			return nil, fmt.Errorf("unexpected URL path (must contain two parts after group '%s'): %s", group, u.Path)
+		_, fileWithSubPath, ok := strings.Cut(u.Path, fmt.Sprintf("%s/", group)) // e.g. "pipeline-runs/foo-zh9gt0.json"
+		if !ok {
+			return nil, fmt.Errorf("unexpected URL path (must contain group '%s'): %s", group, u.Path)
 		}
-		fileWithSubPath := urlPathParts[1] // e.g. "pipeline-runs/foo-zh9gt0.json"
 		if !strings.Contains(fileWithSubPath, "/") {
 			return nil, fmt.Errorf("unexpected URL path (must contain a subfolder after the commit SHA): %s", fileWithSubPath)
 		}
 		aritfactName := path.Base(fileWithSubPath) // e.g. "pipeline-runs"
 		artifactType := path.Dir(fileWithSubPath)  // e.g. "foo-zh9gt0.json"
-		artifactsSubPath := filepath.Join(artifactsDir, artifactType)
-		if _, err := os.Stat(artifactsSubPath); os.IsNotExist(err) {
```

Comment:
I suppose that os.Stat() with a relative path would look for the file in the cwd. This is likely not a desirable feature here, or?

Reply:
If I am not mistaken the code before does change the working directory so this works. But I agree that this is not how things should be. But the PR here does not change this behaviour and I would like to avoid touching it in here.